File: DI/DI-Dagger-Java/src/main/java/DI/persona/modules/PersonasFileModule.java
package DI.persona.modules;

import dagger.Binds;
import dagger.Module;
import model.persona.services.IPersonaStorage;
import model.persona.services.PersonaFileStorage;

import javax.inject.Singleton;

@Module
public interface PersonasFileModule {
    @Singleton
    @Binds
    IPersonaStorage bindPersonaDBStorage(PersonaFileStorage impl);
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/DI/personas/modules/PersonasFileModule.kt
package DI.personas.modules

import dagger.Binds
import dagger.Module
import models.personas.services.IPersonaStorage
import models.personas.services.PersonaFileStorage
import javax.inject.Singleton


@Module
interface PersonasFileModule {
    @Singleton
    @Binds
    fun bindPersonaDBStorage(impl: PersonaFileStorage): IPersonaStorage
}
